### ISSUE TITLE
Gradle build errors: "requires a newer version of Maven. in org.eclipse.jdt:org.eclipse.jdt.annotation:2.2.700"

### ISSUE THREAD

**Issue by dsyer** [jstachio/jstachio#236]
Gradle projects seem to be experiencing this:

```
$ ./gradlew compileJava

> Task :compileJava UP-TO-DATE
Errors occurred while building effective model from /home/dsyer/.gradle/caches/modules-2/files-2.1/org.eclipse.jdt/org.eclipse.jdt.annotation/2.2.700/7b8a6d168e3785f5fba06dc27203969b7f08e7ad/org.eclipse.jdt.annotation-2.2.700.pom:
        'modelVersion' of '4.0' is newer than the versions supported by this version of Maven: [4.0.0]. Building this project requires a newer version of Maven. in org.eclipse.jdt:org.eclipse.jdt.annotation:2.2.700
```

It may be non-fatal but it's definitely distracting. The origin is jstachio:

```
$ ./gradlew dependencyInsight --dependency org.eclipse.jdt.annotation 

> Task :dependencyInsight
Errors occurred while building effective model from /home/dsyer/.gradle/caches/modules-2/files-2.1/org.eclipse.jdt/org.eclipse.jdt.annotation/2.2.700/7b8a6d168e3785f5fba06dc27203969b7f08e7ad/org.eclipse.jdt.annotation-2.2.700.pom:
        'modelVersion' of '4.0' is newer than the versions supported by this version of Maven: [4.0.0]. Building this project requires a newer version of Maven. in org.eclipse.jdt:org.eclipse.jdt.annotation:2.2.700
org.eclipse.jdt:org.eclipse.jdt.annotation:2.2.700
  Variant compile:
    | Attribute Name                 | Provided | Requested    |
    |--------------------------------|----------|--------------|
    | org.gradle.status              | release  |              |
    | org.gradle.category            | library  | library      |
    | org.gradle.libraryelements     | jar      | classes      |
    | org.gradle.usage               | java-api | java-api     |
    | org.gradle.dependency.bundling |          | external     |
    | org.gradle.jvm.environment     |          | standard-jvm |
    | org.gradle.jvm.version         |          | 17           |

org.eclipse.jdt:org.eclipse.jdt.annotation:2.2.700
\--- io.jstach:jstachio:1.3.2
     \--- io.jstach:jstachio-spring:1.3.2
          \--- io.jstach:jstachio-spring-webmvc:1.3.2
               \--- io.jstach:jstachio-spring-boot-starter-webmvc:1.3.2
                    \--- compileClasspath
```

Is there a newer version of that jar we could use?

**Comment by agentgt** [comment]
Some day the eclipse annotation jar will be replaced by JSpecify.

To be honest that looks like a Gradle bug where it cannot figure out `4.0.0 = 4.0` but I will need to look into it some more.

I have used Gradle many times to compile and have not run across that issue before.

**Comment by dsyer** [comment]
Ironically it turns out to be a Maven bug, not a Gradle bug per se, but triggered by the use of Maven libraries in the Spring dependency management Gradle tooling. So if you use the latest start.spring.io and add JStachio you will see these errors.

I see you found the same symptoms in the JDT tools as well. The actual origin is here: https://github.com/eclipse-platform/eclipse.platform/issues/180 (and no sign of a fix yet despite it being a screw up).

**Comment by agentgt** [comment]
@dsyer  This is now fixed but I'm going to wait a little before releasing. In the meantime others can manually use the latest 2.2.800 which was merged in #285 

Speaking of which it is not a true dependency as it is only needed for compile of JStachio so another option for others is to manually exclude but I believe that might still breaks gradle in the older 2.2.700 version.
